src/core/handle.ml

```diff
@@ -276,9 +276,15 @@ let handle_cmd : sig_state -> p_command -> sig_state * proof_data option =
         | P_config_unif_rule(h)   ->
             (* Approximately same processing as rules without SR checking. *)
             let pur = (scope_rule ss h).elt in
+            let rhs =
+              Bindlib.unbox (Bindlib.bind_mvar pur.pr_vars pur.pr_rhs)
+            in
+            let (xs, rhs) = Bindlib.unmbind rhs in
+            let rhs = Unif_rule.prod_obj rhs in
```

Comment:
Is it necessary to apply prod_obj to the RHS?

Reply:
> @gabrielhdt Would it be difficult to modify the trees and the matching algorithm to do matching on products as we already do matching on abstractions? Then, the functions obj/unobj could be implemented by simply adding/removing #prod without doing any lifting/unboxing: Prod(a,b) would simply be transformed into Appl(Symb(#prod),Abst(a,b)), or even Appl(Symb(#prod),Prod(a,b)). Perhaps we could even get rid of obj/unobj by applying he tree of #prod in the case Prod.

If I understand well, you want to be able to consider `Prod` as a tree symbol to match on it and use a `#prod` symbol to hold a tree. But if we allow matching on products, a `#prod` symbol should be useless since the tree is hold by `#equiv`.

So I think matching on products can be implemented, and it would even allow to not use any `#prod` symbol.

Reply:
No indeed